### PR TITLE
feat: add value of env and app in app context

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
add value of env and app in app context

Fixes # (issue)

### CODE REVIEW

File: src/components/app/details/AppHeader.tsx [hunk 1]
```diff
@@ -84,8 +87,13 @@ export const AppHeader = ({
         currentPathname.current = location.pathname
     }, [location.pathname])
 
+    useEffect(() => {
+        setCurrentAppName(appName)
```
Comment (vivek-devtron) [inline src/components/app/details/AppHeader.tsx:91]: Please re verify if it is required

Comment (AbhishekA1509) [reply src/components/app/details/AppHeader.tsx:91]: Verified and for safety added appName in dependency

File: src/components/app/list-new/AppList.tsx [hunk 2]
```diff
@@ -1015,6 +1017,7 @@ export default function AppList({ isSuperAdmin, appListCount, isArgoInstalled }:
                             setAppCount={setAppCount}
                             updateDataSyncing={updateDataSyncing}
                             isArgoInstalled={isArgoInstalled}
+                            setCurrentAppName={setCurrentAppName}
```
Comment (eshankvaish) [inline src/components/app/list-new/AppList.tsx:1020]: Can we not consume the context directly in the component? Check the GlobalConfigurationProvider for usage with class components

File: src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx [hunk 3]
```diff
@@ -150,8 +152,17 @@ export default function EnvCDDetails({ filteredAppIds }: AppGroupDetailDefaultTy
             return
         }
         if (triggerId === triggerDetail?.id) {
+            const appliedFilters = triggerHistory.get(triggerId)?.appliedFilters ?? []
+            const appliedFiltersTimestamp = triggerHistory.get(triggerId)?.appliedFiltersTimestamp
+            const promotionApprovalMetadata = triggerHistory.get(triggerId)?.promotionApprovalMetadata
```
Comment (eshankvaish) [inline src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx:157]: Suggestion: Use destructuring. DRY out triggerHistory.get(triggerId)

**Review by eshankvaish — APPROVED** [review]

File: src/components/app/list-new/AppList.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/app/list-new/AppList.tsx:1020]: We are consuming it with custom hook 🪝  so wanted to keep it same

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**